src/chart/generateCategoricalChart.tsx

```diff
@@ -1343,23 +1349,28 @@ export const generateCategoricalChart = ({
       ];
     }
 
-    inRange(x: number, y: number): any {
+    inRange(x: number, y: number, scale = 1): any {
       const { layout } = this.props;
 
+      const [scaledX, scaledY] = [x / scale, y / scale];
```

Comment:
what happens if these result in NaN?

Reply:
I can make another check here too, but I assumed I dealt with that in the parent scope, before passing the arg here. as this is internal function that is not exposed so I did not do a check again, but I can add it if you think it's better that way

Reply:
Nah it should be fine. I do feel though like this function could just be totally taken out of this file and probably have some different, more targeted tests added to it. Probably an improvement for another time though